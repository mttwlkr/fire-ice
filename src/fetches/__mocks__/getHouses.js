export const getHouses = jest.fn()