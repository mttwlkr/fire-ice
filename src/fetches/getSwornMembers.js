export const getSwornMembers = async (houses) => {

  const promises = houses.map(async(house) => {
    
    const swornMembers = house.swornMembers.map(async(API) => {
      const response = await fetch(API)
      const swornMember = await response.json()
      return swornMember
    })

    return await Promise.all(swornMembers)
  })

  return Promise.all(promises)
}