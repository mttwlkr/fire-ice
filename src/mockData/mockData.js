export const mockHousesArray = [
  { 
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    currentLord: "https://www.anapioficeandfire.com/api/characters/681",
    diedOut: "",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words: "",
  },
  { 
    ancestralWeapons: ["Lady Forlorn"],
    cadetBranches: [],
    coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
    currentLord: "https://www.anapioficeandfire.com/api/characters/681",
    diedOut: "",
    founded: "",
    founder: "https://www.anapioficeandfire.com/api/characters/256",
    heir: "https://www.anapioficeandfire.com/api/characters/677",
    name: "House Corbray of Heart's Home",
    overlord: "https://www.anapioficeandfire.com/api/houses/7",
    region: "",
    seats: ["Heart's Home"],
    swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
    titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
    url: "https://www.anapioficeandfire.com/api/houses/81",
    words: "",
  }  
]

const mockHousesAction = {houses: [
    { 
      ancestralWeapons: ["Lady Forlorn"],
      cadetBranches: [],
      coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
      currentLord: "https://www.anapioficeandfire.com/api/characters/681",
      diedOut: "",
      founded: "",
      founder: "https://www.anapioficeandfire.com/api/characters/256",
      heir: "https://www.anapioficeandfire.com/api/characters/677",
      name: "House Corbray of Heart's Home",
      overlord: "https://www.anapioficeandfire.com/api/houses/7",
      region: "",
      seats: ["Heart's Home"],
      swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
      titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
      url: "https://www.anapioficeandfire.com/api/houses/81",
      words: "",
    },
    { 
      ancestralWeapons: ["Lady Forlorn"],
      cadetBranches: [],
      coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
      currentLord: "https://www.anapioficeandfire.com/api/characters/681",
      diedOut: "",
      founded: "",
      founder: "https://www.anapioficeandfire.com/api/characters/256",
      heir: "https://www.anapioficeandfire.com/api/characters/677",
      name: "House Corbray of Heart's Home",
      overlord: "https://www.anapioficeandfire.com/api/houses/7",
      region: "",
      seats: ["Heart's Home"],
      swornMembers: (8) ["https://www.anapioficeandfire.com/api/characters/255", "https://www.anapioficeandfire.com/api/characters/256", "https://www.anapioficeandfire.com/api/characters/447", "https://www.anapioficeandfire.com/api/characters/654", "https://www.anapioficeandfire.com/api/characters/677", "https://www.anapioficeandfire.com/api/characters/681", "https://www.anapioficeandfire.com/api/characters/871", "https://www.anapioficeandfire.com/api/characters/1182"],
      titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
      url: "https://www.anapioficeandfire.com/api/houses/81",
      words: "",
    }  
  ]
}

export const mockCardWithSwornMembers = { 
  ancestralWeapons: ["Lady Forlorn"],
  cadetBranches: [],
  coatOfArms: "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
  currentLord: "https://www.anapioficeandfire.com/api/characters/681",
  diedOut: "",
  founded: "",
  founder: "https://www.anapioficeandfire.com/api/characters/256",
  heir: "https://www.anapioficeandfire.com/api/characters/677",
  name: "House Corbray of Heart's Home",
  overlord: "https://www.anapioficeandfire.com/api/houses/7",
  region: "",
  seats: ["Heart's Home"],
  swornMembers: [
    {url: "https://www.anapioficeandfire.com/api/characters/255", name: "Corwyn Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/256", name: "Corwyn Corbray", gender: "Male", culture: "Andal", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/447", name: "Gwayne Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/654", name: "Lucas Corbray", gender: "Male", culture: "", born: ""},
    {url: "https://www.anapioficeandfire.com/api/characters/677", name: "Lyn Corbray", gender: "Male", culture: "", born: "At Heart's Home"},
    {url: "https://www.anapioficeandfire.com/api/characters/681", name: "Lyonel Corbray", gender: "Male", culture: "", born: "In 260 AC (roughly), at Heart's Home"},
    {url: "https://www.anapioficeandfire.com/api/characters/871", name: "Rhaena Targaryen", gender: "Female", culture: "", born: "In 116 AC, at Pentos"},
    {url: "https://www.anapioficeandfire.com/api/characters/1182", name: "Baelish", gender: "Male", culture: "Braavosi", born: ""}
  ],
  titles: (2) ["Lord of Heart's Home", "King of the Fingers (historical)"],
  url: "https://www.anapioficeandfire.com/api/houses/81",
  words: "",
}


