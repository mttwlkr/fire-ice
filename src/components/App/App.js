import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { fakeAction } from '../../actions';
import { getHouses } from '../../fetches/getHouses.js'

class App extends Component {

  async componentDidMount() {
    const houses = await getHouses()
    console.log(houses)
  }

  render() {
    return (
      <div className='App'>
        <div className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <h2>Welcome to Westeros</h2>
          <button onClick={() => {
            this.props.fakeAction();
            alert(this.props.fake);
          }}> FAKE ACTION</button>
        </div>
        <div className='Display-info'>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  fake: shape({ fake: string }),
  fakeAction: func.isRequired
};

const mapStateToProps = ({ fake }) => ({ fake });
const mapDispatchToProps = dispatch => ({ fakeAction:() => dispatch(fakeAction())});

export default connect(mapStateToProps, mapDispatchToProps)(App);
