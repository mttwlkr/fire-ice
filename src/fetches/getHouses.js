import { getSwornMembers } from './getSwornMembers.js';

export const getHouses = async () => {
  try {
    const response = await fetch('http://localhost:3001/api/v1/houses')
    const houses = await response.json()
    const swornMembers = await getSwornMembers(houses)
    const housesWithMembers = await houses.map((house, index) => {
      house.swornMembers = swornMembers[index]
      return house
    })
    return housesWithMembers
  } catch (error) {
    throw error
  }
}
