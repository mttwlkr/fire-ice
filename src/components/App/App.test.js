import { App, mapStateToProps, mapDispatchToProps } from './App.js';
import React from 'react';
import { shallow } from 'enzyme';
import { getHouses } from '../../fetches/getHouses';
import { mockHousesArray, mockHousesAction } from '../../mockData/mockData.js'

jest.mock('../../fetches/getHouses') 

describe('App', () => {

  let wrapper;
  let mockHousesToStore = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<App 
      addHousesToStore={mockHousesToStore}
      houses={mockHousesArray}
    />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('componentDidMount should invoke getHouses, addHousesToStore', () => {
    expect(mockHousesToStore).toHaveBeenCalled()
    expect(getHouses).toHaveBeenCalled()
  })

  it('should mapStateToProps correctly', () => {
    const mockState = {houses: ["Ive never seen Game of Thrones"]}
    const mapped = mapStateToProps(mockState)
    expect(mapped.houses).toEqual(mockState.houses)
  })

  it('should mapDispatchToProps correctly', () => {
    const mockDispatch = jest.fn()
    const mapped = mapDispatchToProps(mockDispatch)
    mapped.addHousesToStore()
    expect(mockDispatch).toHaveBeenCalled()
  })
})