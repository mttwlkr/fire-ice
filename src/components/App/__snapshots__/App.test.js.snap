// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addHousesToStore={[Function]}
    houses={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": undefined,
              "titles": undefined,
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
            Object {
              "ancestralWeapons": Array [
                "Lady Forlorn",
              ],
              "cadetBranches": Array [],
              "coatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              "currentLord": "https://www.anapioficeandfire.com/api/characters/681",
              "diedOut": "",
              "founded": "",
              "founder": "https://www.anapioficeandfire.com/api/characters/256",
              "heir": "https://www.anapioficeandfire.com/api/characters/677",
              "name": "House Corbray of Heart's Home",
              "overlord": "https://www.anapioficeandfire.com/api/houses/7",
              "region": "",
              "seats": Array [
                "Heart's Home",
              ],
              "swornMembers": undefined,
              "titles": undefined,
              "url": "https://www.anapioficeandfire.com/api/houses/81",
              "words": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h2>
                    Welcome to Westeros
          </h2>
</div>,
        <div
          className="Display-info"
>
          <Connect(Component) />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h2>
              Welcome to Westeros
</h2>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to Westeros",
            },
            "ref": null,
            "rendered": "Welcome to Westeros",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Component) />,
          "className": "Display-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h2>
                        Welcome to Westeros
            </h2>
</div>,
          <div
            className="Display-info"
>
            <Connect(Component) />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h2>
                Welcome to Westeros
</h2>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to Westeros",
              },
              "ref": null,
              "rendered": "Welcome to Westeros",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Component) />,
            "className": "Display-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
